src/lib.rs

```diff
@@ -192,21 +185,44 @@ pub fn build(config: &Config) -> Result<(), Box<std::error::Error>> {
 
     document.data = Some(PrimaryData::Single(Box::new(krate)));
 
-    let serialized = serde_json::to_string(&document)?;
+    Ok(document)
+}
 
-    let mut file = File::create(json_path)?;
-    file.write_all(serialized.as_bytes())?;
+pub fn build(config: &Config, artifacts: &[&str]) -> Result<(), Box<std::error::Error>> {
```

Comment:
I think this would be good. That way only the `Config` is passed in to build and it doesn't seem like artifacts is used elsewhere besides here